env/iasConfig.go

```diff
@@ -46,17 +52,103 @@ func GetIASConfig() (*Identity, error) {
+	if err != nil {
+		return nil, fmt.Errorf("cannot read service directory '%s' for identity service: %w", secretPath, err)
+	}
+	identities := []Identity{}
+	for _, instanceBound := range instancesBound {
+		if !instanceBound.IsDir() {
+			continue
+		}
+		serviceInstancePath := path.Join(secretPath, instanceBound.Name())
+		instanceSecretFiles, err := ioutil.ReadDir(serviceInstancePath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read service instance directory '%s' for '%s' service instance '%s': %w", serviceInstancePath, iasServiceName, instanceBound.Name(), err)
+		}
+		instanceSecretsJSON, err := readCredentialsFileToJSON(serviceInstancePath, instanceSecretFiles)
+		if instanceSecretsJSON == nil || err != nil {
+			instanceSecretsJSON, err = readSecretFilesToJSON(serviceInstancePath, instanceSecretFiles)
+			if err != nil {
+				return nil, err
+			}
+		}
+		identity := Identity{}
+		if err := json.Unmarshal(instanceSecretsJSON, &identity); err != nil {
+			return nil, fmt.Errorf("cannot unmarshal json content in directory '%s' for '%s' service instance: %w", serviceInstancePath, iasServiceName, err)
+		}
+		identities = append(identities, identity)
+	}
+	return identities, nil
+}
+
+func readCredentialsFileToJSON(serviceInstancePath string, instanceSecretFiles []fs.FileInfo) ([]byte, error) {
+	for _, instanceSecretFile := range instanceSecretFiles {
+		if !instanceSecretFile.IsDir() && instanceSecretFile.Name() == iasSecretKeyDefault {
+			serviceInstanceCredentialsPath := path.Join(serviceInstancePath, instanceSecretFile.Name())
+			credentials, err := ioutil.ReadFile(serviceInstanceCredentialsPath)
+			if err != nil {
+				return nil, fmt.Errorf("cannot read content from '%s': %w", serviceInstanceCredentialsPath, err)
+			}
+			if json.Valid(credentials) {
+				return credentials, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func readSecretFilesToJSON(serviceInstancePath string, instanceSecretFiles []fs.FileInfo) ([]byte, error) {
+	instanceCredentialsMap := make(map[string]interface{})
+	for _, instanceSecretFile := range instanceSecretFiles {
+		if instanceSecretFile.IsDir() {
+			continue
+		}
+		serviceInstanceSecretPath := path.Join(serviceInstancePath, instanceSecretFile.Name())
+		var secretContent []byte
+		secretContent, err := ioutil.ReadFile(serviceInstanceSecretPath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read secret file '%s' from '%s': %w", instanceSecretFile.Name(), serviceInstanceSecretPath, err)
+		}
+		if instanceSecretFile.Name() == "domains" {
+			var domains []string
+			if err := json.Unmarshal(secretContent, &domains); err != nil {
+				return nil, fmt.Errorf("cannot unmarshal content of secret file '%s' from '%s': %w", instanceSecretFile.Name(), serviceInstanceSecretPath, err)
+			}
+			instanceCredentialsMap[instanceSecretFile.Name()] = domains
+		} else if instanceSecretFile.Size() > 0 {
+			instanceCredentialsMap[instanceSecretFile.Name()] = string(secretContent)
```

Comment:
How about using `Unmarshal` here as well. E.g.
```
var v interface{} 
err := json.Unmarshal(secretContent, v)
if err != nil {..}
instanceCredentialsMap[instanceSecretFile.Name()] = v
```
This way, we don't have to have any knowledge about IAS' credential structure and non-string credentials (e.g. []string) are handled properly. This allows us to get rid of the `domains` special case as well.

Reply:
good idea, i already tried it... but it fails with error "invalid character 'c' looking for beginning of value"
reason JSON without double qouted keys it's not JSON at all.

Found a tip to use yaml instead of json parser but this has issues with characters "]" or "," which are valid for secrets. So i still need a fallback.

Reply:
You are right. This is a problem with the BTP service operator. There is no way to distinguish, whether `["asdf"]` is a string that contains special characters or a list with a single string. Let's follow https://github.com/SAP/sap-btp-service-operator/issues/79 and see whether this will be improved.

Reply:
Interesting that you also stumbled upon this. Could you perhaps comment in the issue so that the colleagues can see this is a problem for multiple teams? (also, fyi, this [probably also affects nested objects](https://github.com/SAP/sap-btp-service-operator/issues/79#issuecomment-914211792))

Also, are you aware of this issue: https://github.com/SAP/sap-btp-service-operator/issues/78

I guess you must have also noticed that the plan information is unavailable? At least with the java security libraries in version `2.10.5` the `CfEnvironment` requires a plan. Passing a JSON in there without a plan causes a NPE.